librubyfmt/src/format.rs

```diff
@@ -460,6 +460,9 @@ pub fn use_parens_for_method_call(
         if ps.current_formatting_context() == FormattingContext::Binary {
             return true;
         }
+        if ps.current_formatting_context() == FormattingContext::IfOp {
```

Comment:
Thanks for this pull request! It's almost exactly what I would have done.

Would you mind collapsing this conditional and the one above to a method on parser state: `ps.current_formatting_context_requires_parens()` and then moving `Binary && IfOp` on to that object?

Reply:
Sounds good 😄 How's this? 90303fc6cb058ebedcc6be8c2d8a1d8e0e28447e

I'm writing Rust with absolutely no experience so it was a bit trial and error 😅 The tests and linter both seem happy though.